shared/js/dataDriver.js

```diff
@@ -360,10 +376,19 @@ loop.DataDriver = function() {
      *  - {Mixed}  value     Stored value for the record.
      */
     _processRecord(key, data) {
-      let dispatchAction;
-      let dispatchExtra = {};
       // XXX akita bug 1274130: Validate data/key/timestamp before processing.
       let [, type, id] = key.match(/^([^!]+)!(.+)$/);
+      try {
+        // First validates data, if no exception is thrown we can dispatch the actions
+        this._validateData(type, data);
+      } catch (exception) {
+        console.error("Validate data error: ", exception);
```

Comment:
We're technically validating not just "data" but the "record" which has the key (including type) and the data (including timestamp and value). ;)